purchase_vendor_promotion/models/stock_warehouse_orderpoint.py

```diff
@@ -12,6 +12,25 @@ class StockWarehouseOrderpoint(models.Model):
     )
     promotion_date_start = fields.Date(compute="_compute_promotion", store=True)
     promotion_date_end = fields.Date(compute="_compute_promotion", store=True)
+    supplier_id = fields.Many2one(
+        compute="_compute_supplier_id", readonly=False, store=True
+    )
+
+    @api.depends("route_id", "route_id.force_vendor_with_best_promotion")
+    def _compute_supplier_id(self):
+        for rec in self:
+            if rec.route_id and rec.route_id.force_vendor_with_best_promotion:
+                allowed_company_ids = rec.allowed_location_ids.mapped("company_id")
```

Comment:
```suggestion
                allowed_company_ids = rec.allowed_location_ids.company_id
```
Since Odoo 13.0, `mapped()` (or equivalent method) is automatically executed when fetching relational fields.

Reply:
you're right, thanks